numpyro/distributions/transforms.py

```diff
@@ -1380,6 +1382,70 @@ def __eq__(self, other):
         return jnp.array_equal(self.transition_matrix, other.transition_matrix)
 
 
+class ZeroSumTransform(Transform):
+    """A transform that constrains an array to sum to zero, adapted from PyMC [1] as described in [2,3]
+
+    **References**
+    [1] https://github.com/pymc-devs/pymc/blob/244fb97b01ad0f3dadf5c3837b65839e2a59a0e8/pymc/distributions/transforms.py#L266
+    [2] https://www.pymc.io/projects/docs/en/stable/api/distributions/generated/pymc.ZeroSumNormal.html
+    [3] https://learnbayesstats.com/episode/74-optimizing-nuts-developing-zerosumnormal-distribution-adrian-seyboldt/
+    """
+
+    def __init__(self, zerosum_axes):
```

Comment:
It might be useful to just have `transform_ndims` as in #1762 because the event dimensions are the last dimensions. Say we have an input tensor with shape `(3, 4, 5, 6)` and `zerosum_axes = (-1, -3)`. Then the event dimension implied by `codomain` is two, but in practice the last three axes are dependent.

Reply:
To make sure I'm interpreting correctly: would the idea be to replace something like `ZeroSumTransform(zerosum_axes=(-2, -1))` with `ZeroSumTransform(transform_ndims=2)` as a way to avoid edge cases since we know the event_shape will always be at the end?

Reply:
Yes, that would align with these two:

https://github.com/pyro-ppl/numpyro/blob/84973a98aafd0ebd9ae936a1b4173587c5cdec1b/numpyro/distributions/transforms.py#L569

https://github.com/pyro-ppl/numpyro/blob/84973a98aafd0ebd9ae936a1b4173587c5cdec1b/numpyro/distributions/transforms.py#L1226-L1230

But I'll defer to @fehiepsi.

Reply:
This also solves that `test_parametrized_transform_eq` was failing (I think due to non-jittable parameters when I was supplying a tuple)

Reply:
calling this resolved,  switched to `transform_ndims` and tests are passing